src/main/kotlin/com/zepben/evolve/database/sqlite/writers/NetworkCIMWriter.kt

```diff
@@ -311,6 +311,14 @@ class NetworkCIMWriter(databaseTables: DatabaseTables) : BaseCIMWriter(databaseT
         description: String
     ): Boolean {
         insert.setNullableString(table.POSTAL_CODE.queryIndex, streetAddress.postalCode)
+        insert.setNullableString(table.PO_BOX.queryIndex, streetAddress.poBox)
+        insert.setNullableString(table.BUILDING_NAME.queryIndex, streetAddress.streetDetail?.buildingName ?: "")
+        insert.setNullableString(table.FLOOR_IDENTIFICATION.queryIndex, streetAddress.streetDetail?.floorIdentification ?: "")
+        insert.setNullableString(table.NAME.queryIndex, streetAddress.streetDetail?.name ?: "")
+        insert.setNullableString(table.NUMBER.queryIndex, streetAddress.streetDetail?.number ?: "")
+        insert.setNullableString(table.SUITE_NUMBER.queryIndex, streetAddress.streetDetail?.suiteNumber ?: "")
+        insert.setNullableString(table.TYPE.queryIndex, streetAddress.streetDetail?.type ?: "")
+        insert.setNullableString(table.DISPLAY_ADDRESS.queryIndex, streetAddress.streetDetail?.displayAddress ?: "")
```

Comment:
No need to use `setNullableString` if you change the colmns to be `NOT_NULL` (`POSTAL_CODE` could also be updated)

The `StreetDetail` fields should be split into a seperate function like `saveTownDetail`

Reply:
`saveTownDetail` is currently responsible for executing the update with `tryExecuteSingleUpdate`. Should I move that call over to `saveStreetAddress` (and possibly rename `saveTownOrStreetDetail` to `insertTownOrStreetDetail`)? Or should I leave it as is and simply call `saveStreetDetail` first?

Reply:
(Possibly breaking change?) New revision moves responsibility of executing update to `saveStreetAddress`, transforming `saveTownDetail` and `saveStreetDetail` to `insertTownDetail` and `insertStreetDetail` respectively.